File: 87 3-D Plot/javascript/3dplot.js
// 3D PLOT
//
// Converted from BASIC to Javascript by Oscar Toledo G. (nanochess)
//
function print(str)
{
	document.getElementById("output").appendChild(document.createTextNode(str));
}

function tab(space)
{
	var str = "";
	while (space-- > 0)
		str += " ";
	return str;
}

function equation(input)
{
	return 30 * Math.exp(-input * input / 100);
}

print(tab(32) + "3D PLOT\n");
print(tab(15) + "CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY\n");

for (x = -30; x <= 30; x += 1.5) {
	l = 0;
	y1 = 5 * Math.floor(Math.sqrt(900 - x * x) / 5);
	str = "";
	for (y = y1; y >= -y1; y -= 5) {
		z = Math.floor(25 + equation(Math.sqrt(x * x + y * y)) - .7 * y);
		if (z > l) {
			l = z;
			while (str.length < z)
				str += " ";
			str += "*";
		}
	}
	print(str + "\n");
}

File: 87 3-D Plot/python/3dplot.py
#!/usr/bin/env python3

# 3D PLOT
#
# Converted from BASIC to Python by Trevor Hobson

import math

def equation(input):
	return 30 * math.exp(-input * input / 100)

print(" " * 32 + "3D PLOT")
print(" " * 15 + "CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY\n\n\n\n")

for x in range (-300, 315, 15):
	x1 = x / 10
	l = 0
	y1 = 5 * math.floor(math.sqrt(900 - x1 * x1) / 5)
	yPlot = [" "] * 80

	for y in range (y1, -(y1 + 5), -5):
		z = math.floor(25 + equation(math.sqrt(x1 * x1 + y * y)) - .7 * y)
		if z > l:
			l = z
			yPlot[z] = "*"
	print("".join(yPlot))